Assets/BowAndArrowImproved/Scripts/Ballon.cs

```diff
@@ -50,6 +50,8 @@ private IEnumerator DieCoroutine(float delay)
 
     private void OnCollisionEnter(Collision other)
     {
-        Die(0);
+        if (other.gameObject.CompareTag("Arrow")){
```

Comment:
create a [SerializeField] for the tag and make it configurable through the editor